### ISSUE TITLE
Errno::EACCES when create and add to Zipfile on Windows

### ISSUE THREAD

**Issue by jaykilleen** [rubyzip/rubyzip#457]
I'm porting a program created in a posix environment over to Windows (unfortunately cannot use wsl at this stage).

I am encountering an error when zipping some csv files. 

```ruby
  def zip_each_csv_file
    csv_files = files csv_directory, ext: '.csv'
    csv_files.each do |csv_file|
      zipfile_name = "#{csv_file}.zip"
      zipfile = File.new(zipfile_name, 'w')
      Zip::File.open(zipfile_name, Zip::File::CREATE) do |zipfile|
        zipfile.add(File.basename(csv_file),csv_file)
        zipfile.close
      end
      copy_zip_file_to_outbox zipfile
      File.delete(zipfile) if File.exist?(zipfile)
      move_csv_file_to_archive csv_file
    end
    clear_archive csv_archive
  end
```

```bash
8: from C:/Users/jay/Projects/project-name/lib/inbox.rb:47:in `block in zip_each_csv_file'
7: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:127:in `open'
6: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:127:in `ensure in open'
5: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:373:in `close'
4: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:349:in `commit'
3: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:452:in `on_success_replace'
2: from C:/Ruby27-x64/lib/ruby/2.7.0/tmpdir.rb:129:in `create'
1: from C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:455:in `block in on_success_replace'
C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/rubyzip-2.3.0/lib/zip/file.rb:455:in `rename': Permission denied @ rb_file_s_rename - (C:/Users/jay/Projects/project-name/db/data/csv/20200918064628_things.csv.zip20200918-7664-1bmjs2f, C:/Users/jay/Projects/project-name/db/data/csv/20200918064628_things.csv.zip) (Errno::EACCES)
```

The error is occuring in file.rb
https://github.com/rubyzip/rubyzip/blob/e5e3f97ec891cb596f1af8950fc9749c99b809e9/lib/zip/file.rb#L455

I read in your docs and via Google that this is to do with the permissions by Windows. 

Do you have an example, or how would I be able to handle this action in Windows? I tried monkey patching some `File.chmod(0777,tmp_filename` just to see if I can get things to work but no luck.

Should this gem only be used for posix systems?

**Comment by hainesr** [comment]
Hi, this gem should be OK for Windows as it makes use of standard ruby IO classes such as `File` everywhere. I'm not sure that any of the current maintainers is a big Windows user though so there may be the odd issue. I don't currently have a native Windows dev environment but am looking at setting this up when time allows.

In the meantime, with the code above, are you opening `zipfile` twice? Once with `File` and once with `Zip::File`? Would that be an issue in Windows?

**Comment by jaykilleen** [comment]
Ah ok! That makes perfect sense. I have made the changes and things are now working. I guess in Unix, I have written the code poorly (thinking I first create the zip and then open it and add files). It did work though. Then when I have moved to Windows, due to the different permission defaults, it has thrown the error because it won't allow changes to `zipfile` after it is created and then opened. 

```ruby
  def zip_each_csv_file
    csv_files = files csv_directory, ext: '.csv'
    csv_files.each do |csv_file|
      zipfile_name = "#{csv_file}.zip"
      Zip::File.open(zipfile_name, Zip::File::CREATE) do |zipfile|
        zipfile.add(File.basename(csv_file),csv_file)
      end
    end
  end
```

This issue can be closed and hopefully if anyone makes a similar mistake to me they stumble upon this issue 👍
